### DISCUSSION TITLE
Fetching messages for user

### DISCUSSION THREAD

**Discussion by vercte** [ScratchAddons/ScratchAddons#4589] [Help] [answered]
Just since I'm new to the scratch API, how do you fetch messages? I looked at the source code for the messages popups there, but I can't find how there.

I know what URL you need to get, but what headers?
Do you need a crsfToken?
What other tokens?

Sorry if this isn't allowed here. I just want to know because I'm making a CLI for Scratch.

**Comment by WorldLanguages** [comment, marked as answer]
https://github.com/ScratchAddons/ScratchAddons/blob/master/libraries/common/message-cache.js#L82

The X-Token can be retrieved by fetching `scratch.mit.edu/session/` with proper cookies.

**Reply by vercte** [reply]
An example of a cookie would be? As I said, I'm pretty new to the API. I'm almost there, but what would the cookie look like?

**Reply by WorldLanguages** [reply]
Well, before getting into messages, you'll need to make logging in work. You need to send a POST request to `scratch.mit.edu/login` and get the `scratchsessionsid` cookie from the `Set-Cookie` response header.
